### PR TITLE
Kubelet decides podStatus

### LINKED ISSUE #3433: Make net container invisible to the end users.
Kubelet now only reports ContainerStatus. PodStatus / PodCondition is observed at master side based on a set of ContainerStatus. There is no other way to get PodIP except net container for master side. 
This is very confusion to the end user, #3427, #3429, ...

Now we are finalizing on v1beta3 API, including PodStatus & ContainerStatus. We can make net container completely invisible to the end users by finishing the following steps:

1) Expose network related errors at API level
2) Generate PodStaus directly from Kubelet

### PR DESCRIPTION
**Pull request by dchen1107**
I ran e2e a while back, and test was passed. But this pr was blocked by a failure in service integration test which I identified a bug in PodCache. But didn't have time to look into it in detail lately. Today, I realize @brendandburns just fixed an issue in PodCache, which sounds like a bug I found. I re-run the integration test after sync, now it passed. Unfortunately, I couldn't ran e2e tests today due to breakage, but want to send it out for an initial peek.

cc/ @bgrant0607 

Next PR is completely hiding net container, which should fix #3433 completely.

### CODE REVIEW

File: pkg/api/conversion.go [hunk 1]
```diff
@@ -59,7 +59,6 @@ func init() {
 			out.UUID = in.UID
 			return nil
 		},
-
```
Comment (bgrant0607) [inline pkg/api/conversion.go]: Please revert this.

File: pkg/api/v1beta1/conversion.go [hunk 2]
```diff
@@ -176,7 +176,6 @@ func init() {
 			}
 			return nil
 		},
-
```
Comment (bgrant0607) [inline pkg/api/v1beta1/conversion.go]: Please revert this.

File: pkg/master/pod_cache.go [hunk 3]
```diff
@@ -187,11 +186,11 @@ func (p *PodCache) computePodStatus(pod *api.Pod) (api.PodStatus, error) {
 		newStatus.Phase = api.PodUnknown
 	} else {
 		newStatus.Info = result.Status.Info
-		newStatus.Phase = getPhase(&pod.Spec, newStatus.Info)
-		if netContainerInfo, ok := newStatus.Info[leaky.PodInfraContainerName]; ok {
-			if netContainerInfo.PodIP != "" {
-				newStatus.PodIP = netContainerInfo.PodIP
-			}
+		newStatus.PodIP = result.Status.PodIP
+		if newStatus.Info == nil {
```
Comment (bgrant0607) [inline pkg/master/pod_cache.go]: This is for before Kubelet has populated the Status? Please add a comment.

File: pkg/master/pod_cache.go [hunk 4]
```diff
@@ -246,67 +245,3 @@ func (p *PodCache) UpdateAllContainers() {
 	}
 	wg.Wait()
 }
-
-// getPhase returns the phase of a pod given its container info.
```
Comment (bgrant0607) [inline pkg/master/pod_cache.go]: YEAH!

**Comment by dchen1107** [comment]
Reverted that unnecessary changes due to debugging. PTAL?

**Comment by bgrant0607** [comment]
Basically LGTM. A few minor issues. I assume getPhase and the tests were copied over.

File: pkg/master/pod_cache.go (hunk 3 above)
Comment (dchen1107) [inline pkg/master/pod_cache.go]: Yes, add the comment.

**Comment by dchen1107** [comment]
Yes, and I am running e2e for now.

**Comment by dchen1107** [comment]
e2e passed:

Ran 20 of 20 Specs in 1397.412 seconds
I0205 16:53:30.972923   29333 driver.go:78] All tests pass
SUCCESS! -- 20 Passed | 0 Failed | 0 Pending | 0 Skipped watchEvents stdout: |

This PR is ready to merge now.

**Pull request merged**